src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs

```diff
@@ -119,6 +119,14 @@ protected virtual void ActivateProperties([NotNull] object controller, [NotNull]
                     }
                 },
                 {
+                    typeof(ITempDataDictionary),
```

Comment:
You can use `[FromServices]` on the controller, and then you don't need to add any code.